include/parallel/parallel_implementation.h

```diff
@@ -1168,6 +1168,10 @@ inline void Communicator::nonblocking_receive_packed_range (const unsigned int s
   // Make the Request::wait() then handle deleting the buffer
   req.add_post_wait_work
     (new Parallel::PostWaitDeleteBuffer<std::vector<buffer_t>>(buffer));
+
+  // The MessageTag should stay registered for the Request lifetime
+  req.add_post_wait_work
+    (new Parallel::PostWaitDereferenceTag(tag));
```

Comment:
This is a straight-up bug fix, right, having nothing to do with the rest of your changes?

Reply:
Correct.